feat: add amazon-q module

DevelopmentCats opened the pull request:
## Description
Adds the `amazon-q` module for automated installation and management of Amazon Q in Coder workspaces.

- **Authentication via a pre-authenticated, base64-encoded tarball is required.**
- Installs Amazon Q with configurable version and folder.
- Supports passing prompts to Amazon Q tasks.
- Reports status directly in the Coder app UI.
- Runs Amazon Q in the background with `tmux` or `screen`.
- Optional pre/post-install scripts and task reporting integration.
- Robust error handling and flexible configuration via Terraform variables.

Credit: @hugodutka (amazon-q module)

DevelopmentCats committed:
feat(amazon-q): add initial module implementation with tests and README

- Introduced the `amazon-q` module, which installs and configures Amazon's AI coding assistant.
- Added `main.tf` for Terraform resource definitions, including variables and local configurations.
- Created `main.test.ts` for testing the module's functionality, ensuring required resources are created.
- Included a comprehensive `README.md` with usage instructions, prerequisites, and examples for running Amazon Q in various configurations.

DevelopmentCats committed:
refactor(amazon-q): remove validation for experiment_auth_tarball variable

- Updated the `experiment_auth_tarball` variable in `main.tf` to remove the validation condition, allowing it to default to null. This change allows the module to build and for the validation to be done on the template side

DevelopmentCats committed:
fix(README): update Amazon Q installation link to the correct developer site

DevelopmentCats committed:
chore(README): fix typo in TODO

DevelopmentCats committed:
fix(amazon-q): remove unused experiment_auth_tarball variable from tests

DevelopmentCats committed:
fix(amazon-q): update default value for experiment_auth_tarball variable

- Changed the default value of the `experiment_auth_tarball` variable in `main.tf` from null to "tarball" to provide a more meaningful default for users, and to allow for validation at the template level

DevelopmentCats commented:
@hugodutka Can you Review this PR for the Amazon-Q Module?

I have tested it and its working perfectly, with some minor changes.
![amazonq-example-better](https://github.com/user-attachments/assets/2e245814-85a3-4f85-9511-2e37e888d5e7)

matifali commented on amazon-q/README.md:
```diff
@@ -0,0 +1,143 @@
+<details>
+<summary><strong>How to generate the Amazon Q auth tarball (step-by-step)</strong></summary>
+
+**1. Install and authenticate Amazon Q on your local machine:**
```
Don't they support an easier more user friendly auth method?

hugodutka replied:
I wish - that’s what’s recommened in the community forums. https://community.aws/content/2uZYCp6BNJJgBaRnw3Nie6i8r0l/putting-amazon-q-developer-in-a-docker-container

hugodutka commented on amazon-q/main.tf:
```diff
@@ -0,0 +1,294 @@
+      exit 1
+    fi
+
+    FULL_PROMPT="$(printf "%s\n\nYour first task is:\n\n%s" "$CODER_MCP_AMAZON_Q_SYSTEM_PROMPT" "$CODER_MCP_AMAZON_Q_TASK_PROMPT")"
```
Task prompt and system prompt should be configurable via module variables, and we should ship a default system prompt that instructs the agent how to do task reporting.

hugodutka commented on amazon-q/README.md:
````diff
@@ -0,0 +1,143 @@
+
+- Run the following command in your terminal (Linux/macOS):
+  ```sh
+  cd ~/.local/share/amazon-q
````
On macOS the path is different, and I haven't tested whether a tarball from macOS will work on Linux. If we document macOS in the README, we must make sure that the instructions are correct.

hugodutka commented on amazon-q/README.md:
```diff
@@ -0,0 +1,143 @@
+- If neither is set, Amazon Q runs in the foreground.
+- For more details, see the [main.tf](./main.tf) source.
+
+<!-- TODO: Add Screenshot of Amazon-Q In Action -->
```
looks like this TODO wasn't handled

hugodutka commented on amazon-q/main.tf:
```diff
@@ -0,0 +1,294 @@
+    fi
+
+    # Run with screen if enabled
+    if [ "${var.experiment_use_screen}" = "true" ]; then
```
Have you tested the module with screen? I only ever ran it with tmux.

DevelopmentCats replied:
Yeah I tested it with screen as well it worked fine in my test but I'll double check when I get to my workstation in about an hour

hugodutka commented on amazon-q/main.tf:
```diff
@@ -0,0 +1,294 @@
+      PREV_DIR="$PWD"
+      TMP_DIR="$(mktemp -d)"
+      cd "$TMP_DIR"
+      curl --proto '=https' --tlsv1.2 -sSf "https://desktop-release.q.us-east-1.amazonaws.com/${var.amazon_q_version}/q-x86_64-linux.zip" -o "q.zip"
```
We should probably do an architecture check here since we only support x64 Linux

matifali replied:
If they have an arm64 asset, let's do that too. I have seen recent PRs of users adding arm64 support to other modules.

matifali commented:
We are moving to https://github.com/coder/regitsry. Please move your PR there. Thanks
More context in #469

DevelopmentCats committed:
chore(amazon-q): add AmazonQ In Action Example Image

DevelopmentCats committed:
feat(amazon-q): enhance prompts and architecture detection

- Added `system_prompt` and `task_prompt` variables
- Updated the script to detect system architecture and download the appropriate version of Amazon Q for x86_64 and arm64.
- Refactored the prompt handling to use a local variable combining system and task prompt

DevelopmentCats committed:
docs(amazon-q): update README with image and variable formatting

- Added a screenshot of Amazon Q in action to enhance visual understanding.
- Removed unverified instructions regarding `zstd` installation for macOS.
- Add variables for System and Task Prompts

DevelopmentCats committed:
fix(amazon-q): rename task_prompt variable to ai_prompt

DevelopmentCats committed:
fix(amazon-q): update README to rename task_prompt to ai_prompt

DevelopmentCats committed:
fix(amazon-q): complete rename task_prompt to ai_prompt in main.tf

DevelopmentCats committed:
fix(amazon-q): update system_prompt to enhance task reporting guidelines.

DevelopmentCats committed:
chore(amazon-q): clean up script by removing commented sections

stirby commented:
Did this get migrated?

matifali commented:
yes to https://github.com/coder/registry/pull/95
